docs/nnx/bridge_guide.md

```diff
@@ -0,0 +1,544 @@
+
+(However, we still keep the whole collection as one class attribute, because Linen modules may have duplicated names over different collections.)
+
+
+```python
+class LinenMultiCollections(nn.Module):
+  out_dim: int
+  def setup(self):
+    self.w = self.param('w', nn.initializers.lecun_normal(), (x.shape[-1], self.out_dim))
+    self.b = self.param('b', nn.zeros_init(), (self.out_dim,))
+    self.count = self.variable('counter', 'count', lambda: jnp.zeros((), jnp.int32))
+
+  def __call__(self, x):
+    if not self.is_initializing():
+      self.count.value += 1
+    y = x @ self.w + self.b
+    self.sow('intermediates', 'dot_sum', jnp.sum(y))
+    return y
+
+x = jax.random.normal(jax.random.key(42), (2, 4))
+model = bridge.lazy_init(bridge.ToNNX(LinenMultiCollections(3), rngs=nnx.Rngs(0)), x)
+print(model.params['w'])        # Of type `nnx.Param` - note this is still under attribute `params`
+print(model.params['b'])        # Of type `nnx.Param`
+print(model.counter['count'])   # Of type `counter` - an auto-created dummy type from the name "counter"
+print(type(model.counter['count']))
+
+y = model(x, mutable=True)              # Linen's `sow()` needs `mutable=True` to trigger
+print(model.intermediates['dot_sum'])   # Of type `nnx.Intermediates`
+```
+
+    Param(
+      value=Array([[ 0.35401407,  0.38010964, -0.20674096],
+             [-0.7356256 ,  0.35613298, -0.5099556 ],
+             [-0.4783049 ,  0.4310735 ,  0.30137998],
+             [-0.6102254 , -0.2668519 , -1.053598  ]], dtype=float32)
+    )
+    Param(
+      value=Array([0., 0., 0.], dtype=float32)
+    )
+    counter(
+      value=Array(0, dtype=int32)
+    )
+    <class 'abc.counter'>
+    (Intermediate(
+      value=Array(6.932987, dtype=float32)
+    ),)
+
+
+You can quickly separate different types of NNX variables apart using `nnx.split`.
+
+This can be handy when you only want to set some variables as trainable.
+
+
+```python
+# Separate variables of different types with nnx.split
+CountType = type(model.counter['count'])
+static, params, counter, the_rest = nnx.split(model, nnx.Param, CountType, ...)
+print('All Params:', list(params['params'].keys()))
+print('All Counters:', list(counter['counter'].keys()))
+print('All the rest (intermediates and RNG keys):', list(the_rest.keys()))
+
+model = nnx.merge(static, params, counter, the_rest)  # You can merge them back at any time
+```
+
+    All Params: ['b', 'w']
+    All Counters: ['count']
+    All the rest (intermediates and RNG keys): ['intermediates', 'rngs']
+
+
+### NNX to Linen
+
+If you define custom NNX variable types, you should register their names with `nnx.register_variable_name_type_pair` so that they go to the desired collections.
+
+
+```python
+class Count(nnx.Variable): pass
+nnx.register_variable_name_type_pair('Count', Count, overwrite=True)
```

Comment:
`Count` would be the automatic name anyway, maybe we can show registering a Linen-style name (no capital, plural)?
```suggestion
nnx.register_variable_name_type_pair('counts', Count, overwrite=True)
```

Reply:
Done.